### PR TITLE
feat(obfuscate columns udf) :Obfuscate columns udf

### PR DESCRIPTION
**Pull request by golankiviti**

### CODE REVIEW

**Review by amirhalatzi — APPROVED** [review]
Very nice! can you pleaser update the Readme with a usage example?

File: src/main/scala/com/yotpo/metorikku/code/steps/InputMatcher.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.yotpo.metorikku.code.steps
+
+case class InputMatcher[K](ks: K*) {
```
Comment (Doe-Ed) [inline src/main/scala/com/yotpo/metorikku/code/steps/InputMatcher.scala:3]: nice!

File: src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ColumnsAreNotPartOfTheOriginalSchemaException.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.yotpo.metorikku.code.steps.obfuscate
+
+case class ColumnsAreNotPartOfTheOriginalSchemaException(columns: Array[String])
```
Comment (Doe-Ed) [inline src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ColumnsAreNotPartOfTheOriginalSchemaException.scala:3]: ColumnsAreNotPartOfTheOriginalSchemaException --> ColumnsNotPartOfOriginalSchema

File: src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ObfuscateColumns.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import com.yotpo.metorikku.code.steps.InputMatcher
+import com.yotpo.metorikku.exceptions.MetorikkuException
+
+object ObfuscateColumns {
+  private val ObfuscateColumnsInputMatcher = InputMatcher("columns", "delimiter", "value", "table")
+
+  def run(ss: org.apache.spark.sql.SparkSession, metricName: String, dataFrameName: String, params: Option[Map[String, String]]): Unit = {
+    val (columns, value, table) = parse(params)
+    val df = ss.table(table)
+    val newDataFrame = obfuscateColumns(df, columns, value)
```
Comment (Doe-Ed) [inline src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ObfuscateColumns.scala:15]: obfuscatedDf

File: src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ObfuscateColumns.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    newDataFrame.createOrReplaceTempView(dataFrameName)
+  }
+
+  private def parse(params: Option[Map[String, String]]): (Array[String], String, String) = {
+    params.get match {
+      case ObfuscateColumnsInputMatcher(columns, delimiter, value, table) => (columns.split(delimiter), value, table)
+      case _ => throw MetorikkuException(
+        """Obfuscate Columns expects the following parameters: columns,
+          delimiter, value and table. one or more of those wasn't supplied""".stripMargin
```
Comment (Doe-Ed) [inline src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ObfuscateColumns.scala:25]: Pandas are advocates of the Oxford comma!

Comment (golankiviti) [reply src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ObfuscateColumns.scala]: https://youtu.be/j64SctPKmqk?t=92

**Comment by golankiviti** [comment]
> Very nice! can you pleaser update the Readme with a usage example?

@amirhalatzi There isn't a region in the Readme for a udf examples.
I could add a Readme under /examples/udf which will be docs for all the udfs, what do oyu think?

File: src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ObfuscateColumns.scala [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    val (columns, value, table) = parse(params)
+    val df = ss.table(table)
+    val obfuscatedDf = obfuscateColumns(df, columns, value)
+
+    obfuscatedDf.createOrReplaceTempView(dataFrameName)
+  }
+
+  private def parse(params: Option[Map[String, String]]): (Array[String], String, String) = {
+    params.get match {
+      case ObfuscateColumnsInputMatcher(columns, delimiter, value, table) => (columns.split(delimiter), value, table)
```
Comment (liebstein) [inline src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ObfuscateColumns.scala:22]: how about a default value to delimiter?

Comment (golankiviti) [reply src/main/scala/com/yotpo/metorikku/code/steps/obfuscate/ObfuscateColumns.scala:22]: I don't think it's necessary

**Review by liebstein — APPROVED** [review]

**Review by Doe-Ed — APPROVED** [review]

**Pull request merged**